File: frameworks/Java/fair/java-kora1-jdbc-repository/src/main/java/io/techempower/benchmark/kora/repository/WorldRepository.java
package io.techempower.benchmark.kora.repository;

import io.techempower.benchmark.kora.model.Fortune;
import io.techempower.benchmark.kora.model.World;
import ru.tinkoff.kora.database.common.annotation.Batch;
import ru.tinkoff.kora.database.common.annotation.Query;
import ru.tinkoff.kora.database.common.annotation.Repository;
import ru.tinkoff.kora.database.jdbc.JdbcRepository;

import java.util.List;

@Repository
public interface WorldRepository extends JdbcRepository {

    @Query("SELECT id, randomnumber FROM world WHERE id = :id")
    World findById(int id);

    @Query("SELECT randomnumber FROM world WHERE id = :id")
    int findRandomNumberById(int id);

    @Query("UPDATE world SET randomnumber = :world.randomNumber WHERE id = :world.id")
    void update(@Batch List<World> world);

    @Query("SELECT id, message FROM fortune")
    List<Fortune> fortunes();
}

File: frameworks/Java/fair/kotlin-kora2-jdbc-repository/src/main/kotlin/io/techempower/benchmark/kora/repository/WorldRepository.kt
package io.techempower.benchmark.kora.repository

import io.techempower.benchmark.kora.model.Fortune
import io.techempower.benchmark.kora.model.World
import io.koraframework.database.common.annotation.Batch
import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcRepository

@Repository
interface WorldRepository : JdbcRepository {

    @Query("SELECT id, randomnumber FROM world WHERE id = :id")
    fun findById(id: Int): World

    @Query("SELECT randomnumber FROM world WHERE id = :id")
    fun findRandomNumberById(id: Int): Int

    @Query("UPDATE world SET randomnumber = :world.randomNumber WHERE id = :world.id")
    fun update(@Batch world: List<World>)

    @Query("SELECT id, message FROM fortune")
    fun fortunes(): List<Fortune>
}
